agimus_controller/agimus_controller/trajectories/sine_wave_cartesian_space.py

```diff
@@ -15,48 +18,117 @@ class SinusWaveCartesianSpace(TrajectoryBase):
 
     def __init__(
         self,
-        sine_wave_params,
+        sine_wave_params: SinWaveParams,
         ee_frame_name,
         w_q,
         w_qdot,
         w_qddot,
         w_robot_effort,
         w_pose,
+        mask=6 * [True],
     ):
-        """Initialize parameters needed for the sine wave in configuration space trajectory."""
+        """Initialize parameters needed for the sine wave in configuration space trajectory.
+
+        Args:
+            sine_wave_params (SinWaveParams): Sine wave parameters: period, amplitude, scale_duration.
+            ee_frame_name (_type_): Name of the end effector frame.
+            w_q (_type_): weight for the robot configuration.
+            w_qdot (_type_): weight for the robot velocity.
+            w_qddot (_type_): weight for the robot acceleration.
+            w_robot_effort (_type_): weight for the robot effort.
+            w_pose (_type_): weight for the end effector pose.
+            mask (_type_, optional): Inverse kinematics DoF mask [x, y, z, roll, pitch, yaw]. Defaults to 6*[True].
+        """
+
         super().__init__(ee_frame_name)
         self.quint_traj = QuinticTrajectory(
             scale_duration=sine_wave_params.scale_duration
         )
-        self.amp = sine_wave_params.amplitude
-        self.w = 2.0 * np.pi / sine_wave_params.period  # pulsation
+        self.amp = np.array(sine_wave_params.amplitude)
+        self.w = np.array(sine_wave_params.pulsation)
         self.w_q = w_q
         self.w_qdot = w_qdot
         self.w_qddot = w_qddot
         self.w_robot_effort = w_robot_effort
         self.w_pose = w_pose
+        self.mask = mask
+        self.ik_q = None
 
-    def get_traj_point_at_t(self, t: np.float64) -> WeightedTrajectoryPoint:
-        quint, _, _ = self.quint_traj.get_value_at_t(t)
-        w = self.w
-        sin_wt = np.sin(w * t)
-        # TODO Implement inv kyn to retrieve correct references
-        self.q = self.q0
-        self.dq = self.dq
-        self.ddq = self.ddq
-        for i in [0, 1]:
-            ee_pose = (
-                self.get_end_effector_pose_from_q(self.q0) + self.amp * quint * sin_wt
+    def initialize(self, pin_model: pin.Model, q0: np.ndarray) -> None:
+        """Initialize the trajectory generator."""
+        super().initialize(pin_model, q0)
+        self.ik_q = self.q0.copy()
+        self.ee_init_pos = self.get_end_effector_pose_from_q_as_se3(self.q0)
+
+    def inverse_kinematics(
+        self,
+        ee_des_pos: pin.SE3,
+        ee_des_vel: np.ndarray,
+        precision=1e-3,
```

Comment:
`1e-3` is a bit too large to my taste. The algo should easily reach a lower value.
```suggestion
        precision: float=1e-5,
```